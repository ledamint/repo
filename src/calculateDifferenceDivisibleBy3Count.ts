export function calculateDifferenceDivisibleBy3Count(
  firstNumber: number,
  secondNumber: number
) {
  if (firstNumber > secondNumber) {
    throw new Error("secondNumber should be more than firstNumber");
  }

  let divisibleBy3Count = 0;

  const isFirstNumberDivisibleBy3 = firstNumber % 3 === 0;
  const isSecondNumberDivisibleBy3 = secondNumber % 3 === 0;

  if (isFirstNumberDivisibleBy3) {
    divisibleBy3Count += 1;
  }

  if (isSecondNumberDivisibleBy3) {
    divisibleBy3Count += 1;
  }

  if (isFirstNumberDivisibleBy3 && isSecondNumberDivisibleBy3) {
    divisibleBy3Count -= 1;
  }

  divisibleBy3Count += Math.floor((secondNumber - firstNumber) / 3);

  return divisibleBy3Count;
}
