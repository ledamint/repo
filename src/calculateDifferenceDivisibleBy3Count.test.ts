import { calculateDifferenceDivisibleBy3Count } from "./calculateDifferenceDivisibleBy3Count";

describe("calculateDifferenceDivisibleBy3Count", () => {
  it("should return error if firstNumber should be more than secondNumber", () => {
    expect(calculateDifferenceDivisibleBy3Count(2, 1)).toEqual(
      "secondNumber should be more than firstNumber"
    );
  });

  it("should return 0 if difference is 0 and arguments are not divisible by 3", () => {
    expect(calculateDifferenceDivisibleBy3Count(1, 1)).toEqual(0);
    expect(calculateDifferenceDivisibleBy3Count(4, 4)).toEqual(0);
    expect(calculateDifferenceDivisibleBy3Count(7, 7)).toEqual(0);
    expect(calculateDifferenceDivisibleBy3Count(-1, -1)).toEqual(0);
  });

  it("should return 1 if difference is 0 and arguments are divisible by 3", () => {
    expect(calculateDifferenceDivisibleBy3Count(3, 3)).toEqual(1);
    expect(calculateDifferenceDivisibleBy3Count(6, 6)).toEqual(1);
    expect(calculateDifferenceDivisibleBy3Count(0, 0)).toEqual(1);
  });

  it("should return 0 if difference is 1 and arguments are not divisible by 3", () => {
    expect(calculateDifferenceDivisibleBy3Count(1, 2)).toEqual(0);
    expect(calculateDifferenceDivisibleBy3Count(4, 5)).toEqual(0);
    expect(calculateDifferenceDivisibleBy3Count(7, 8)).toEqual(0);
    expect(calculateDifferenceDivisibleBy3Count(-8, -7)).toEqual(0);
  });

  it("should return 1 if difference is 2 and first argument is divisible by 3", () => {
    expect(calculateDifferenceDivisibleBy3Count(3, 5)).toEqual(1);
    expect(calculateDifferenceDivisibleBy3Count(6, 8)).toEqual(1);
    expect(calculateDifferenceDivisibleBy3Count(9, 11)).toEqual(1);
    expect(calculateDifferenceDivisibleBy3Count(-3, -1)).toEqual(1);
    expect(calculateDifferenceDivisibleBy3Count(0, 2)).toEqual(1);
  });

  it("should return 1 if difference is 2 and second argument is divisible by 3", () => {
    expect(calculateDifferenceDivisibleBy3Count(1, 3)).toEqual(1);
    expect(calculateDifferenceDivisibleBy3Count(4, 6)).toEqual(1);
    expect(calculateDifferenceDivisibleBy3Count(7, 9)).toEqual(1);
    expect(calculateDifferenceDivisibleBy3Count(-5, -3)).toEqual(1);
    expect(calculateDifferenceDivisibleBy3Count(-2, 0)).toEqual(1);
  });

  it("should return 1 if difference is 3 and arguments are not divisible by 3", () => {
    expect(calculateDifferenceDivisibleBy3Count(1, 4)).toEqual(1);
    expect(calculateDifferenceDivisibleBy3Count(2, 5)).toEqual(1);
    expect(calculateDifferenceDivisibleBy3Count(7, 10)).toEqual(1);
    expect(calculateDifferenceDivisibleBy3Count(-7, -4)).toEqual(1);
    expect(calculateDifferenceDivisibleBy3Count(-1, 2)).toEqual(1);
  });

  it("should return 2 if difference is 3 and arguments are divisible by 3", () => {
    expect(calculateDifferenceDivisibleBy3Count(3, 6)).toEqual(2);
    expect(calculateDifferenceDivisibleBy3Count(6, 9)).toEqual(2);
    expect(calculateDifferenceDivisibleBy3Count(9, 12)).toEqual(2);
    expect(calculateDifferenceDivisibleBy3Count(-6, -3)).toEqual(2);
    expect(calculateDifferenceDivisibleBy3Count(-3, 0)).toEqual(2);
  });

  it("should return 2 if difference is 4 and first argument is divisible by 3", () => {
    expect(calculateDifferenceDivisibleBy3Count(3, 7)).toEqual(2);
    expect(calculateDifferenceDivisibleBy3Count(6, 10)).toEqual(2);
    expect(calculateDifferenceDivisibleBy3Count(9, 13)).toEqual(2);
    expect(calculateDifferenceDivisibleBy3Count(-6, -2)).toEqual(2);
    expect(calculateDifferenceDivisibleBy3Count(0, 4)).toEqual(2);
  });

  it("should return 2 if difference is 4 and second argument is divisible by 3", () => {
    expect(calculateDifferenceDivisibleBy3Count(2, 6)).toEqual(2);
    expect(calculateDifferenceDivisibleBy3Count(5, 9)).toEqual(2);
    expect(calculateDifferenceDivisibleBy3Count(8, 12)).toEqual(2);
    expect(calculateDifferenceDivisibleBy3Count(-7, -3)).toEqual(2);
    expect(calculateDifferenceDivisibleBy3Count(-4, 0)).toEqual(2);
  });

  it("should return 2 if difference is 5 and first argument is divisible by 3", () => {
    expect(calculateDifferenceDivisibleBy3Count(3, 8)).toEqual(2);
    expect(calculateDifferenceDivisibleBy3Count(6, 11)).toEqual(2);
    expect(calculateDifferenceDivisibleBy3Count(9, 14)).toEqual(2);
    expect(calculateDifferenceDivisibleBy3Count(-6, -1)).toEqual(2);
    expect(calculateDifferenceDivisibleBy3Count(-3, 2)).toEqual(2);
    expect(calculateDifferenceDivisibleBy3Count(0, 5)).toEqual(2);
  });

  it("should return 2 if difference is 5 and second argument is divisible by 3", () => {
    expect(calculateDifferenceDivisibleBy3Count(1, 6)).toEqual(2);
    expect(calculateDifferenceDivisibleBy3Count(4, 9)).toEqual(2);
    expect(calculateDifferenceDivisibleBy3Count(7, 12)).toEqual(2);
    expect(calculateDifferenceDivisibleBy3Count(-8, -3)).toEqual(2);
    expect(calculateDifferenceDivisibleBy3Count(-2, 3)).toEqual(2);
    expect(calculateDifferenceDivisibleBy3Count(-5, 0)).toEqual(2);
  });

  it("should return 3 if difference is 6 and arguments are divisible by 3", () => {
    expect(calculateDifferenceDivisibleBy3Count(3, 9)).toEqual(3);
    expect(calculateDifferenceDivisibleBy3Count(6, 12)).toEqual(3);
    expect(calculateDifferenceDivisibleBy3Count(9, 15)).toEqual(3);
    expect(calculateDifferenceDivisibleBy3Count(-9, -3)).toEqual(3);
    expect(calculateDifferenceDivisibleBy3Count(-6, 0)).toEqual(3);
    expect(calculateDifferenceDivisibleBy3Count(0, 6)).toEqual(3);
    expect(calculateDifferenceDivisibleBy3Count(-3, 3)).toEqual(3);
  });
});
