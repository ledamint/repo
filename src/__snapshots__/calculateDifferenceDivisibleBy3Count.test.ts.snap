// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculateDifferenceDivisibleBy3Count should return error if firstNumber should be more than secondNumber: secondNumber should be more than firstNumber 1`] = `"secondNumber should be more than firstNumber"`;
